### PR TITLE
Fix bug when using timestamps for dispatcher task IDs

### PR DESCRIPTION
**Pull request by aaronchongth**
### Fixed bug

Fixes https://github.com/open-rmf/rmf_ros2/issues/252. This issue only shows up when the dispatcher node is not running with sim time due to casting issues when the double gets larger.

To recreate the issue, set `use_sim_time` to `false` when launching the dispatcher node in 
https://github.com/open-rmf/rmf_demos/blob/main/rmf_demos/launch/common.launch.xml#L60

### Fix applied

* using `std::lround` instead of `std::round)`

### CODE REVIEW

**Review by luca-della-vedova — APPROVED** [review]

**Comment by mxgrey** [comment]
We should probably wait for approval from @Yadunund before merging so we don't disturb the release process that's going on.

**Comment by Yadunund** [comment]
Yep kindly hold for a day or two more. Shall update here.

**Comment by luca-della-vedova** [comment]
> Yep kindly hold for a day or two more. Shall update here.

I think we should be OK right?

**Pull request merged**
